lib/classifier-reborn/lsi.rb

```diff
@@ -60,7 +66,11 @@ def needs_rebuild?
     #
     def add_item( item, *categories, &block )
       clean_word_hash = Hasher.clean_word_hash((block ? block.call(item) : item.to_s), @language)
-      @items[item] = ContentNode.new(clean_word_hash, *categories)
+      @items[item] = if @cache_node_vectors
+        CachedContentNode.new(clean_word_hash, *categories)
+      else
+        ContentNode.new(clean_word_hash, *categories)
+      end
```

Comment:
what do you think about a `new_content_node()` method that abstracts this creation?